bot/exts/moderation/silence.py

```diff
@@ -92,88 +113,230 @@ async def _async_init(self) -> None:
+                overwrite = channel.overwrites_for(self._everyone_role)
+                manual = overwrite.send_messages is False or overwrite.add_reactions is False
+
+            # Send fail message to muted channel or voice chat channel, and invocation channel
+            if manual:
+                await self.send_message(MSG_UNSILENCE_MANUAL, msg_channel, channel, alert_target=False)
+            else:
+                await self.send_message(MSG_UNSILENCE_FAIL, msg_channel, channel, alert_target=False)
+
         else:
-            await channel.send(MSG_UNSILENCE_SUCCESS)
+            await self.send_message(MSG_UNSILENCE_SUCCESS, msg_channel, channel, alert_target=True)
 
-    async def _set_silence_overwrites(self, channel: TextChannel) -> bool:
+    async def _set_silence_overwrites(self, channel: TextOrVoiceChannel, *, kick: bool = False) -> bool:
         """Set silence permission overwrites for `channel` and return True if successful."""
-        overwrite = channel.overwrites_for(self._everyone_role)
-        prev_overwrites = dict(send_messages=overwrite.send_messages, add_reactions=overwrite.add_reactions)
+        # Get the original channel overwrites
+        if isinstance(channel, TextChannel):
+            role = self._everyone_role
+            overwrite = channel.overwrites_for(role)
+            prev_overwrites = dict(send_messages=overwrite.send_messages, add_reactions=overwrite.add_reactions)
 
+        else:
+            role = self._verified_voice_role
+            overwrite = channel.overwrites_for(role)
+            prev_overwrites = dict(speak=overwrite.speak)
+            if kick:
+                prev_overwrites.update(connect=overwrite.connect)
+
+        # Stop if channel was already silenced
         if channel.id in self.scheduler or all(val is False for val in prev_overwrites.values()):
             return False
 
-        overwrite.update(send_messages=False, add_reactions=False)
-        await channel.set_permissions(self._everyone_role, overwrite=overwrite)
+        # Set new permissions, store
+        overwrite.update(**dict.fromkeys(prev_overwrites, False))
+        await channel.set_permissions(role, overwrite=overwrite)
         await self.previous_overwrites.set(channel.id, json.dumps(prev_overwrites))
 
         return True
 
-    async def _schedule_unsilence(self, ctx: Context, duration: Optional[int]) -> None:
+    @staticmethod
+    async def _get_afk_channel(guild: Guild) -> VoiceChannel:
+        """Get a guild's AFK channel, or create one if it does not exist."""
+        afk_channel = guild.afk_channel
+
+        if afk_channel is None:
+            overwrites = {
+                guild.default_role: PermissionOverwrite(speak=False, connect=False, view_channel=False)
+            }
+            afk_channel = await guild.create_voice_channel("mute-temp", overwrites=overwrites)
+            log.info(f"Failed to get afk-channel, created temporary channel #{afk_channel} ({afk_channel.id})")
```

Comment:
Would this log be more fitting around the uses of this method? The method itself doesn't enforce it being temporary

Reply:
This method was only intended to be used within the scope of this cog, so I hadn't considered any problems with that log. I see what you mean though, but it would add useless complexity if we had to convey to the caller that a channel was created. I opted instead to remove the temporary part so the function is not assuming intentions. d9ff7575